misk-hibernate/src/main/kotlin/misk/hibernate/HibernateModule.kt

```diff
@@ -106,6 +106,20 @@ class HibernateModule(
       )
     }).asSingleton()
 
+    if (readerQualifier != null) {
+      val readerTransacterKey = Transacter::class.toKey(readerQualifier)
+      bind(readerTransacterKey).toProvider(object : Provider<Transacter> {
+        @Inject lateinit var executorServiceFactory: ExecutorServiceFactory
+        override fun get(): RealTransacter = RealTransacter(
+            qualifier = readerQualifier,
+            sessionFactoryProvider = readerSessionFactoryProvider!!,
+            readerSessionFactoryProvider = readerSessionFactoryProvider,
```

Comment:
...I wish we could do away with this entirely, but too many codebases are now using replicaRead. We now have two official ways to access the read only replica and one is going through what you think is a writer annotated DB.

Could you update the comment on replicaRead{} to steer people away from the leaky abstraction?

Reply:
How strongly we want to steep people away from `readReplica`?

Reply:
I feel pretty strongly about it because I think of a transacter as a map 1:1 to a particular DB state. Having a transacter map to either a RO or a RW DB instance based on a method has a strong code smell.

Reply:
If we feel strongly about it, we should add a Deprecated annotation with a message on `readReplica` that suggests the path forward using this less leaky approach.

Reply:
We need to do some additional work to make this work for TiDB, and Vitess. I'm not going to add the `Deprecated` annotation for now.

Reply:
The reason we have a `replicaRead` method is that on TiDB and Vitess it's possible to do replica reads with the same connection so there's no need to maintain separate connection pools for writes and replica reads. I also made it possible to use the `replicaRead` syntax on MySQL.

I agree 100% that we should settle on a single way to do this and then support that on both Aurora and TiDB. I have a preference but I don't feel strongly which approach we pick as long as there is one way to do it, not two. :-)

(Personally I prefer the succinctness and explicitness of the `replicaRead` syntax since it behaves differently from a `transaction` method. I prefer if semantic differences has a syntactical representation. Using a `transaction` method for both doesn't seem to fulfill that. But again, I don't feel strongly, either way is fine as long as there is just one.)